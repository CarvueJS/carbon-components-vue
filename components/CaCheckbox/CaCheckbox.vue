<template>
  <div data-checkbox class="bx--form-item bx--checkbox-wrapper">
    <input :id="itemId" class="bx--checkbox" type="checkbox" :value="itemValue" name="checkbox">
    <label :for="itemId" class="bx--checkbox-label">
      <span class="bx--checkbox-appearance">
        <svg class="bx--checkbox-checkmark" width="12" height="9" viewBox="0 0 12 9" fill-rule="evenodd">
          <path d="M4.1 6.1L1.4 3.4 0 4.9 4.1 9l7.6-7.6L10.3 0z"></path>
        </svg>
      </span>
      <slot></slot>
    </label>
  </div>
</template>

<script>
//  import { initCheckbox } from 'carbon-components';

  export default {
    name: 'ca-checkbox',
    props: {
      itemId: {
        type: String,
        defalut: '',
      },
      itemValue: {
        type: String,
        defalut: '',
      },
    },
    data() {
      return {
        id: null,
        value: null,
      };
    },
    mounted() {
//      initCheckbox(document.querySelector[data-checkbox])
    },
  };
</script>

<style></style>
