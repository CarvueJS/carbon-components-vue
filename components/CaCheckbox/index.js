import CaCheckbox from './CaCheckbox';

CaCheckbox.install = (Vue) => {
  Vue.component(CaCheckbox.name, CaCheckbox);
};

export default CaCheckbox;
