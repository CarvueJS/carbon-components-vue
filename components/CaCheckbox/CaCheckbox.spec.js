import { shallow } from 'vue-test-utils';
import CaCheckbox from './CaCheckbox.vue';

describe('CaCheckbox', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaCheckbox, {
      propsData: {
        inputValue: []
      }
    });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('click', () => {
    const model = ['a']
    const wrapper = shallow(CaCheckbox, {
      attachToDocument: true,
      propsData: {
        inputValue: model,
        itemId: 'a',
        itemValue: 'a'
      }
    });
    wrapper.find('input').element.checked = true
    expect(model).toEqual(
      expect.arrayContaining(['a']),
    );
  });
})
