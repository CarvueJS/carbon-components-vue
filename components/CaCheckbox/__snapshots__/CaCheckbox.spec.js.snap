// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaCheckbox Mathes snapshot 1`] = `
<div data-checkbox="" class="bx--form-item bx--checkbox-wrapper">
  <input id="" type="checkbox" name="checkbox" aria-checked="false" class="bx--checkbox" value="">
  <label for="" class="bx--checkbox-label"><span class="bx--checkbox-appearance"><svg width="12" height="9" viewBox="0 0 12 9" fill-rule="evenodd" class="bx--checkbox-checkmark"><path d="M4.1 6.1L1.4 3.4 0 4.9 4.1 9l7.6-7.6L10.3 0z"></path></svg></span> </label>
</div>
`;
